### ISSUE TITLE
com.android.support v4 update breaks the plugin for Android

### ISSUE THREAD

**Issue by JuanjoPP** [pwlin/cordova-plugin-file-opener2#199]
Hello,

The problem is described here:
https://stackoverflow.com/questions/49162538/cordova-build-android-unable-to-find-attribute-androidfontvariationsettings-a

Seems like leaving this line:
`<framework src="com.android.support:support-v4:+" />` 

in the plugin.xml file left the plugin exposed to a breaking change in the com.android.support library, which causes the build to fail.

Setting it to version 27.1.0 before adding the platform seems to work locally, so if someone could please  merge pull request #198 over here it would save us all a bit of a headache.

**Comment by natanielpaiva** [comment]
Please, please! I alson need...

**Comment by JuanjoPP** [comment]
I've set up a fork in the meantime. using `cordova plugin add https://github.com/JuanjoPP/cordova-plugin-file-opener2` and readding the platform should do the trick.

**Comment by cao1240225118** [comment]
Will the author fix it? @JuanjoPP What changes have you made in your branch？It can be said concretely！

**Comment by JuanjoPP** [comment]
https://github.com/JuanjoPP/cordova-plugin-file-opener2/commit/96e394315b926e7787e30cf348d1b9dfd5a653c4

**Comment by MauroRomer** [comment]
@JuanjoPP After I applied your fork I am receiving this erro:

Error: /Projects/Cordova/WalterTuche - TESTE/platforms/android/gradlew: Command failed with exit code 1 Error output:
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':transformClassesWithJarMergingForDebug'.
> com.android.build.api.transform.TransformException: java.util.zip.ZipException: duplicate entry: android/support/v13/view/DragAndDropPermissionsCompat.class

Can you help me?

Tks

**Comment by JuanjoPP** [comment]
https://github.com/phonegap/phonegap-plugin-push/issues/2229#issuecomment-369510785

After a cursory google search, It seems that you may be using a plugin
forcing cordova android support to 26, which crashes with this. You'll
probably have to use your own fork/local plugin for this one, or use the
original and force build.gradle to resolve to whatever version you're using.

El 10 mar. 2018 6:34 a. m., "MauroRomer" <notifications@github.com>
escribió:

@JuanjoPP <https://github.com/juanjopp> After I applied your fork I am
receiving this erro:

Error: /Projects/Cordova/WalterTuche - TESTE/platforms/android/gradlew:
Command failed with exit code 1 Error output:
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

FAILURE: Build failed with an exception.

   - What went wrong:
   Execution failed for task ':transformClassesWithJarMergingForDebug'.

com.android.build.api.transform.TransformException:
java.util.zip.ZipException: duplicate entry: android/support/v13/view/
DragAndDropPermissionsCompat.class

Can you help me?

Tks

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/pwlin/cordova-plugin-file-opener2/issues/199#issuecomment-372004781>,
or mute the thread

**Comment by devjoks** [comment]
same problem here ... can you update the plugin.xml ?? 
use:
```
<framework src="com.android.support:support-v4:27+" />
```
instead of:
```
<framework src="com.android.support:support-v4:+" />
```

**Comment by zbarbuto** [comment]
I was able to fix the error by installing [this plugin](https://github.com/floatinghotpot/cordova-plugin-android-support-v4) (found from [this comment](https://github.com/phonegap/phonegap-plugin-barcodescanner/issues/616#issuecomment-371602548)) :

`cordova plugin add cordova-android-support-gradle-release --variable ANDROID_SUPPORT_VERSION=24.+`

**Comment by JoeMeeks** [comment]
If you're using [@ionic-native/file-opener](https://ionicframework.com/docs/native/file-opener/) run `ionic cordova plugin add cordova-android-support-gradle-release --variable ANDROID_SUPPORT_VERSION=24.+`. This will fix Android build errors in Ionic Pro.
Thanks @zbarbuto! Great Scott!

**Comment by shnist** [comment]
we've just released a new version, 2.1.0, that allows you to optionally specify an android support version at installation
`cordova plugin add cordova-plugin-file-opener2  --variable ANDROID_SUPPORT_VERSION={required version}` 
this should hopefully resolve all the issues around this.
